src/charm.py

```diff
@@ -356,6 +357,9 @@ def set_secret(self, scope: Scopes, key: str, value: Optional[str]) -> Optional[
             return self.remove_secret(scope, key)
 
         peers = self.model.get_relation(PEER)
+        if not peers:
+            return None
```

Comment:
```suggestion
        if not (peers := self.model.get_relation(PEER)):
            return None
```
You can do a walrus check to do the if and the assignment in the same go.